server/controllers/authController.ts

```diff
@@ -1,11 +1,57 @@
-import catchAsync from '../utils/catchAsync';
+import { Request, Response, NextFunction } from "express";
+import catchAsync from "../utils/catchAsync";
+import * as msal from "@azure/msal-node";
 
-const login = catchAsync((req, res, next) => {
-  req;
-  res;
-  next;
-});
-
-export default {
-  login,
+const config = {
+  auth: {
+    clientId: `${process.env.CLIENT_ID}`,
+    authority: `${process.env.AUTHORITY_URL}`,
+    clientSecret: `${process.env.CLIENT_SECRET}`,
+  },
+  system: {
+    loggerOptions: {
+      loggerCallback(_loglevel: any, message: string, _containsPii: any) {
+        console.log(message);
+      },
+      piiLoggingEnabled: false,
+      logLevel: msal.LogLevel.Verbose,
+    },
+  },
 };
+
+const loggingSession = new msal.ConfidentialClientApplication(config);
+
+export const login = catchAsync(
+  async (req: Request, res: Response, next: NextFunction) => {
+    const authCodeUrlParameters = {
+      scopes: ["user.read"],
+      redirectUri: "http://localhost:8080/",
+    };
+
+    const response = await loggingSession.getAuthCodeUrl(authCodeUrlParameters);
+    console.log(response);
+    res.status(200).json({
+      status: "success",
+      session: response,
+    });
+  }
+);
+
+export const getToken = catchAsync(
+  async (req: Request, res: Response, next: NextFunction) => {
+    const tokenRequest = {
+      code: req.query.code as string,
```

Comment:
what is code?

Reply:
Code is result of the first step of authentication using something code "azure authentiaction by code flow" I think